### ISSUE TITLE
Requested neighbor pair method does not exist when using the table/omp pair style

### ISSUE THREAD

**Issue by luitfc** [lammps/lammps#1140]
## Summary

I have been testing the performance of different acceleration packages of LAMMPS for a system using the hybrid/overlay style (`pair_style hybrid/overlay buck/coul/long 8.0 12.0 table linear 23001 pppm`) on our Intel Xeon E5-2680v4 HPC. The case that uses the user-intel package with buck/coul/long/intel and table/omp failed, giving the following error: _Requested neighbor pair method does not exist_. 

## Type of Issue

Bug Report

## Detailed Description (Enhancement Suggestion)
Testing for four cases has been performed. Three worked as expected and one failed, see the details below:

- Case 1 (worked as expected): use the standard LAMMPS without any acceleration packages

```
pair_style hybrid/overlay buck/coul/long 8.0 12.0 table linear 23001 pppm
```

- Case 2 (worked as expected): use the user-omp package

```
suffix omp
package omp 1
pair_style hybrid/overlay buck/coul/long 8.0 12.0 table linear 23001 pppm
```

- Case 3 (worked as expected): use the user-intel package, Intel pair style for buck/coul/long and standard table potential

```
package intel 0 omp 1
pair_style hybrid/overlay buck/coul/long/intel 8.0 12.0 table linear 23001 pppm
```

- Case 4 (failed): use the user-intel package, Intel pair style for buck/coul/long and omp style for the table potential

```
package omp 1
package intel 0 omp 1
pair_style hybrid/overlay buck/coul/long/intel 8.0 12.0 table/omp linear 23001 pppm
```

The error message for case 4 is as follows:

> Hybrid pair style last /omp style table/omp
> Last active /omp style is pair_style table/omp
> ERROR: Requested neighbor pair method does not exist (../neighbor.cpp:739)
> Last command: run 300000

## LAMMPS Version (Bug Report)

The development version: 18Sep2018

## Expected Behavior (Bug Report)
As per the official documentation of LAMMPS, the tabulated potential supports the user-omp style. Case 4 above should work.

## Actual Behavior (Bug Report)
Case 4 gives the following error:

> Hybrid pair style last /omp style table/omp
> Last active /omp style is pair_style table/omp
> ERROR: Requested neighbor pair method does not exist (../neighbor.cpp:739)
> Last command: run 300000

**Comment by akohlmey** [comment]
@sjplimp do you have any idea, where this is coming from? USER-INTEL or USER-OMP or something else?

**Comment by mkanski** [comment]
The error occurs because by default OMP package enforces that all neighbor lists (even those which are requested by non-OMP commands) are multi-threaded (see the snippet below). Intel pair styles require Intel neighbor lists and, since there are no neighbor lists which are both multi-threaded and Intel, the error occurs.
The short workaround is to disable multi-threaded neighbor lists: `package omp 1 neigh no`, however it will reduce efficiency. I guess a better solution would be to move the request for multi-threaded neighbor lists from fix_omp.cpp to individual OMP commands as it is done for the Intel package.

https://github.com/lammps/lammps/blob/cdea8968c26f157f2012b6bdd6a7df872d46d5b1/src/USER-OMP/fix_omp.cpp#L309-L323

**Comment by akohlmey** [comment]
@mkanski thanks for the analysis. the intention of the FixOMP::set_neighbor_omp() function is to enable multi-threaded neighbor list builds *always*, even, if there is no corresponding USER-OMP style. there are some pair styles outside of USER-OMP, that would benefit from it, and even if the pair style is not multi-threaded itself, there will be some speedup from doing a multi-threaded neighbor list build. 

so the solution has to be to make this function smarter. let me think about this for a bit. it is clear now, that the bug is within the USER-OMP package, i.e. it is my fault.

**Comment by akohlmey** [comment]
@luitfc please note, that `USER-INTEL` does not really support hybrid pair styles. it will only work in the case of `hybrid/overlay` where *all* interactions are processed by the `USER-INTEL` threaded pair style.

Beyond that, i have just submitted a pull request which updates the code in fix omp and fix intel to avoid flagging a neighbor list request as USER-INTEL and USER-OMP at the same time. this should alleviate the immediate problem. i would still recommend to be careful and it seems a bit strange to me, that you are using hybrid/overlay (i.e. adding interactions) and not plain hybrid (i.e. using different pair styles for subsets of interactions). could it be, that you are using hybrid/overlay to work around the issue, that USER-INTEL doesn't support skip lists, and thus is incompatible with normal hybrid?

**Comment by luitfc** [comment]
@akohlmey Thanks for helping with solving this issue.
For our problem, we need to use the `hybrid/overlay` style because the two potentials are superposed in an additive fashion to compute the interaction between atoms. A bit interesting thing is about case 3 above where I set `pair_style hybrid/overlay buck/coul/long/intel 8.0 12.0 table linear 23001 pppm`. The former potential uses its `USER_INTEL` threaded variant and the later doesn't. It worked. My question is how these two got combined to process interactions without any compatibility issue, or the interactions are calculated separately for each potential?

Following on your question about to use `hybrid/overlay` for a `hybrid` system to work round the `USER-INTEL` skip lists issue, is that OK to do so just for getting some performance gain?

**Comment by luitfc** [comment]
@akohlmey thanks for your explanation. Much clear now. 

BTW, I can close this issue now, or wait until the next release?

**Comment by akohlmey** [comment]
you can close it, if you like, but it will be automatically closed, as soon as the corresponding pull request is merged. GitHub has this neat feature, that you embed remarks like `closes #1140` or `fixes #1140` into a commit message or a pull request description and then it will automatically close the corresponding issue. That is how we make a connection between issues and pull requests, and can look them up later, if needed. whether an issue is manually closed or not, doesn't interfere with it.
